04-Multitasking/context_switch.S

```diff
@@ -19,14 +19,32 @@ activate:
 	mrs ip, psr
 	push {r4, r5, r6, r7, r8, r9, r10, r11, ip, lr}
 
-	/* switch to process stack */
-	msr psp, r0
-	mov r0, #3
-	msr control, r0
-	isb
 
 	/* load user state */
-	pop {r4, r5, r6, r7, r8, r9, r10, r11, lr}
+	ldmia r0!, {r4, r5, r6, r7, r8, r9, r10, r11, lr}
+
+	msr psp, r0
+
+	/* check the situaion and determine the path to return */
+	mov r0, #0xfffffff0
+	cmp lr, r0                      @ lr - 0xfffffff0
+
+	/* switch to process stack */
+	ittt ls                         @ `ls` condition means unsinged lower or the same
+	movls r0, #3
+	msrls control, r0
+	isbls
+
+	/* jump to user task */
+	bx lr/* check the situaion and determine the path to return */
+	mov r0, #0xfffffff0
+	cmp lr, r0                      @ lr - 0xfffffff0
+
+	/* switch to process stack */
+	ittt ls                         @ `ls` condition means unsinged lower or the same
```

Comment:
I can understand the previous one, but what's wrong with this section?  Should I change "/* switch to process stack */" to "@switch to process stack" ? I just follow the original coding-style.